presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java

```diff
@@ -125,4 +126,38 @@ public void testJsonFileBasedFunction()
         assertQuery("SELECT json.test_schema.map(json.test_schema.array_constructor(linenumber), json.test_schema.array_constructor(linenumber)) FROM lineitem", "SELECT map(array_constructor(linenumber), array_constructor(linenumber)) FROM lineitem");
         assertQuery("SELECT json.test_schema.map_entries(json.test_schema.map(json.test_schema.array_constructor(linenumber), json.test_schema.array_constructor(linenumber))) FROM lineitem", "SELECT map_entries(map(array_constructor(linenumber), array_constructor(linenumber))) FROM lineitem");
     }
+
+    @Test
+    public void testAggregationCompanionFunction()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty("table_writer_merge_operator_enabled", "false")
+                .setCatalogSessionProperty("hive", "collect_column_statistics_on_write", "false")
+                .setCatalogSessionProperty("hive", "optimized_partition_update_serialization_enabled", "false")
+                .build();
+        getQueryRunner().execute(session,
+                "CREATE TABLE avg_partial_states AS ( "
+                    + "SELECT orderpriority, cast(json.test_schema.avg_partial(shippriority) as ROW(sum DOUBLE, count BIGINT)) as states "
+                    + "FROM orders "
+                    + "GROUP BY orderstatus, orderpriority "
+                + ")");
+
+        // Test group-by aggregation.
+        assertQuery(
+                "SELECT orderpriority, json.test_schema.avg_merge_extract_double(states) FROM avg_partial_states GROUP BY orderpriority",
+                "SELECT orderpriority, avg(shippriority) FROM orders GROUP BY orderpriority");
+        assertQuery(
+                "SELECT orderpriority, json.test_schema.avg_extract_double(json.test_schema.avg_merge(states)) FROM avg_partial_states GROUP BY orderpriority",
+                "SELECT orderpriority, avg(shippriority) FROM orders GROUP BY orderpriority");
+
+        // Test global aggregation.
+        assertQuery(
+                "SELECT json.test_schema.avg_merge_extract_double(states) FROM avg_partial_states",
+                "SELECT avg(shippriority) FROM orders");
+        assertQuery(
+                "SELECT json.test_schema.avg_extract_double(json.test_schema.avg_merge(states)) FROM avg_partial_states",
+                "SELECT avg(shippriority) FROM orders");
+
+        getQueryRunner().execute("DROP TABLE IF EXISTS avg_partial_states");
```

Comment:
If the assertion fails (or any exception is raised before `DROP TABLE` instruction, the table will not be dropped; 
if this is acceptable, no changes is required, otherwise, I would use try-finally to make sure the table is dropped at tall times.